tests/mocha/test_helpers.js

```diff
@@ -81,6 +84,86 @@ function createEventsFireStub() {
   return stub;
 }
 
+/**
+ * Shared setup method that sets up fake timer for clock so that pending
+ * setTimeout calls can be cleared in test teardown along with other common
+ * stubs. Should be called in setup of outermost suite using
+ * sharedTestSetup.call(this).
+ * @param {Object<string, boolean>} options Options to enable/disable setup
+ *    of certain stubs.
+ */
+function sharedTestSetup(options = {}) {
+  this.sharedSetupCalled_ = true;
+  // Sandbox created for greater control when certain stubs are cleared.
+  this.sharedSetupSandbox_ = sinon.createSandbox();
+  this.clock = this.sharedSetupSandbox_.useFakeTimers();
```

Comment:
Creating this sandbox allows for other tests to use `sinon.restore()` and not affect the stubbing of the clock.
This is important for preventing leakage of `setTimeout` calls between tests.